### PR TITLE
adding docstrings to appelb.py

### PR DESCRIPTION
**Pull request by joshuaroot**
* adding docstrings to appelb.py

### CODE REVIEW

File: c7n/resources/appelb.py [hunk 1]
```diff
@@ -220,7 +308,17 @@ def __call__(self, alb):
 @filters.register('healthcheck-protocol-mismatch')
 class AppELBHealthCheckProtocolMismatchFilter(Filter,
                                               AppELBTargetGroupFilterBase):
-    """
+    """Filter to pull ELB with mismatched health check protocols
```
Comment (kapilt) [inline c7n/resources/appelb.py]: this doesn't explain what that means.

File: c7n/resources/appelb.py [hunk 2]
```diff
@@ -96,18 +96,39 @@ def _remove_appelb_tags(albs, session_factory, tag_keys):
 
 @filters.register('security-group')
 class SecurityGroupFilter(net_filters.SecurityGroupFilter):
+    """Filter ELB by security group"""
```
Comment (kapilt) [inline c7n/resources/appelb.py]: this doc should come from base class

File: c7n/resources/appelb.py [hunk 3]
```diff
@@ -96,18 +96,39 @@ def _remove_appelb_tags(albs, session_factory, tag_keys):
...
 
     RelatedIdsExpression = "SecurityGroups[]"
 
 
 @filters.register('subnet')
 class SubnetFilter(net_filters.SubnetFilter):
+    """Filter ELB by associated subnet"""
```
Comment (kapilt) [inline c7n/resources/appelb.py]: this doc should come from base class

File: c7n/resources/appelb.py [hunk 4]
```diff
@@ -240,7 +335,20 @@ def _healthcheck_protocol_mismatch(alb):
...
+    :example:
+
+        .. code-block: yaml
+
+            policies:
+              - name: appelb-target-group
+                resource: app-elb
+                filters:
+                  - type: target-group
+                    key: length([?Protocol=='HTTPS'])
```
Comment (kapilt) [inline c7n/resources/appelb.py]: this expression is overly complex, vs a simple key: Protocol value: HTTPS

File: c7n/resources/appelb.py [hunk 5]
```diff
@@ -203,7 +272,20 @@ def initialize(self, albs):
...
+    :example:
+
+        .. code-block: yaml
+
+            policies:
+              - name: appelb-https-listener
+                resource: app-elb
+                filters:
+                  - type: listener
+                    key: length([?Protocol=='HTTPS'])
```
Comment (kapilt) [inline c7n/resources/appelb.py]: this expression is overly complex for the purpose or documentation, vs a simple key: Protocol value: HTTPS

File: c7n/resources/appelb.py (hunk 4 above)
Comment (kapilt) [inline c7n/resources/appelb.py]: also rather than copy and paste a filter, use an attribute unique to target group.

**Comment by joshuaroot** [comment]
docstring examples contained are valid. removed a few that i have been unable to validate. will review and try to determine what im doing wrong with 'listener' and 'target-group' and make a new pr for them at another time

**Pull request merged**